webapp/src/app.tsx

```diff
@@ -1677,148 +1683,168 @@ export class ProjectView
             h.cloudUserId = this.getUserProfile()?.id;
         }
 
-        return compiler.newProjectAsync()
-            .then(() => h.backupRef ? workspace.restoreFromBackupAsync(h) : Promise.resolve())
-            .then(() => pkg.loadPkgAsync(h.id))
-            .then(() => {
-                if (!this.state || this.state.header != h) {
-                    this.showPackageErrorsOnNextTypecheck();
+        try {
+            await compiler.newProjectAsync();
+
+            if (h.backupRef) {
+                await workspace.restoreFromBackupAsync(h);
+            }
+            await pkg.loadPkgAsync(h.id);
+
+            if (!this.state || this.state.header != h) {
+                this.showPackageErrorsOnNextTypecheck();
+            }
+            simulator.setDirty();
+
+            await this.loadTutorialJresCodeAsync();
```

Comment:
Do you need to call ` await compiler.applyUpgradesAsync()` here?

Reply:
whoops, good catch!